internal/fabric/client/client_ccp.go

```diff
@@ -0,0 +1,244 @@
+	mspImpl "github.com/hyperledger/fabric-sdk-go/pkg/msp"
+	log "github.com/sirupsen/logrus"
+)
+
+// the ccpClientWrapper implements the RPCClient interface using the fabric-sdk-go implementation
+// based on the static network description provided via the CCP yaml
+type ccpClientWrapper struct {
+	channelClient   *channel.Client
+	channelProvider context.ChannelProvider
+	signer          *msp.IdentityIdentifier
+}
+
+// defined to allow mocking in tests
+type channelCreator func(context.ChannelProvider) (*channel.Client, error)
+
+type ccpRPCWrapper struct {
+	txTimeout           int
+	sdk                 *fabsdk.FabricSDK
+	cryptoSuiteConfig   core.CryptoSuiteConfig
+	userStore           msp.UserStore
+	idClient            IdentityClient
+	ledgerClientWrapper *ledgerClientWrapper
+	eventClientWrapper  *eventClientWrapper
+	channelCreator      channelCreator
+	// one channel client per channel ID, per signer ID
+	channelClients map[string](map[string]*ccpClientWrapper)
+}
+
+func newRPCClientFromCCP(configProvider core.ConfigProvider, txTimeout int, userStore msp.UserStore, idClient IdentityClient, ledgerClientWrapper *ledgerClientWrapper, eventClientWrapper *eventClientWrapper) (RPCClient, error) {
+	configBackend, _ := configProvider()
+	cryptoConfig := cryptosuite.ConfigFromBackend(configBackend...)
+	identityConfig, err := mspImpl.ConfigFromBackend(configBackend...)
+	if err != nil {
+		return nil, errors.Errorf("Failed to load identity configurations: %s", err)
+	}
+	clientConfig := identityConfig.Client()
+	if clientConfig.CredentialStore.Path == "" {
+		return nil, errors.Errorf("User credentials store path is empty")
+	}
+
+	log.Infof("New gRPC connection established")
+	w := &ccpRPCWrapper{
+		sdk:                 ledgerClientWrapper.sdk,
+		cryptoSuiteConfig:   cryptoConfig,
+		userStore:           userStore,
+		idClient:            idClient,
+		ledgerClientWrapper: ledgerClientWrapper,
+		eventClientWrapper:  eventClientWrapper,
+		channelClients:      make(map[string]map[string]*ccpClientWrapper),
+		channelCreator:      createChannelClient,
+		txTimeout:           txTimeout,
+	}
+	return w, nil
+}
+
+func (w *ccpRPCWrapper) Init(channelId, signer, chaincodeName, method string, args []string) (*TxReceipt, error) {
+	log.Tracef("RPC [%s:%s:%s] --> %+v", channelId, chaincodeName, method, args)
+
+	signerID, result, txStatus, err := w.sendTransaction(channelId, signer, chaincodeName, method, args, true)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Tracef("RPC [%s:%s:%s] <-- %+v", channelId, chaincodeName, method, result)
+	return newReceipt(result, txStatus, signerID), err
+}
+
+func (w *ccpRPCWrapper) Invoke(channelId, signer, chaincodeName, method string, args []string) (*TxReceipt, error) {
+	log.Tracef("RPC [%s:%s:%s] --> %+v", channelId, chaincodeName, method, args)
+
+	signerID, result, txStatus, err := w.sendTransaction(channelId, signer, chaincodeName, method, args, false)
+	if err != nil {
+		return nil, err
```

Comment:
For errors here and in other places, should we log the error as well, along with returning it back to the caller?